File: algorithms/0001/main.go
package p0001

func twoSum(nums []int, target int) []int {
	indexMap := make(map[int]int, len(nums))
	for i, num := range nums {
		if j, ok := indexMap[target-num]; ok {
			return []int{j, i}
		} else {
			indexMap[num] = i
		}
	}
	return nil
}

File: algorithms/0001/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const map = {}
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i]
    const diff = target - num
    if (map[diff] !== undefined) {
      return [map[diff], i]
    } else {
      map[num] = i
    }
  }
}

module.exports = twoSum
